pkg/registry/graph.go

```diff
@@ -0,0 +1,326 @@
+package registry
+
+import (
+	"fmt"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Type declarations
+
+// Type identifies the type of registry element a Node refers to
+type Type int
+
+const (
+	Workflow Type = iota
+	Chain
```

Comment:
@bbguimaraes is this the weird Go thing where this `const` is not of the type you'd expect?

Reply:
Usually yes, but `iota` seems to makes it work: https://play.golang.org/p/2VVuVZZZJz5.